### DISCUSSION TITLE
Internal apps through authentik with traefik

### DISCUSSION THREAD

**Discussion by kekec777** [goauthentik/authentik#7260] [General]
Hi everyone 
Im kinda new to authentik and im trying to setup my apps to guide through authentik.
My installation is in docker and all my container is label correct so that part is working :)

But i have some entries in traefiks config.yml like this, so none docker installation:
`
  routers:
    filebrowser:
      entryPoints:
        - "websecure"
      rule: "Host(filebrowser.MYDOMAIN.COM)"
      middlewares:
        - https-redirect

  services:
    filebrowser:
      loadBalancer:
        servers:
          - url: "http://192.168.1.100:3670"
        passHostHeader: true

  middlewares:
    https-redirect:
      redirectScheme:
        scheme: https
    auth:
      forwardauth:
        address: http://authentikserver:9000/outpost.goauthentik.io/auth/traefik
        trustForwardHeader: true
        authResponseHeaders:
          - X-authentik-username
          - X-authentik-groups
          - X-authentik-email
          - X-authentik-name
          - X-authentik-uid
          - X-authentik-jwt
          - X-authentik-meta-jwks
          - X-authentik-meta-outpost
          - X-authentik-meta-provider
          - X-authentik-meta-app
          - X-authentik-meta-version
`

And it seems no matter what i do, i cannot let this filebrowser.MYDOMAIN.COM to get through authentik
Any ideas or guides to accomplish this? 

Any input is appreciated :)

**Comment by jcsomerville** [comment]
I've been trying to get this to work for forever as well. 

To my understanding it's on Filebrowser's end. I cannot for the life of me get Filebrowser to switch from JSON authentication to Proxy. Even though they do [allow it](hTDTzJXQ=4UFPWKLDR3O3GGFFB37KK5LMLFCWQT6G2MRYNJHT63CBT7IA6LOIMV3LLU4RAYJOIJMN7GNZTJDTQ43UDTQEY72FYXCVT7D5X3AQYTY).

My dynamic config for traefik looks more like this:

http:
  middlewares:
    authentik:
      forwardAuth:
        address: http://192.168.1.188:9010/outpost.goauthentik.io/auth/traefik
        trustForwardHeader: true
        authResponseHeaders:
          - X-authentik-username
          - X-authentik-groups
          - X-authentik-email
          - X-authentik-name
          - X-authentik-uid
          - X-authentik-jwt
          - X-authentik-meta-jwks
          - X-authentik-meta-outpost
          - X-authentik-meta-provider
          - X-authentik-meta-app
          - X-authentik-meta-version
          - authorization
          
  routers:
    filebrowser:
      middlewares:
        - authentik
      entryPoints:
        - web-secure 
      rule: Host(`shared.DOMAIN.com`)
      tls:
        certresolver: myresolver
      service: filebrowser

   services:
    filebrowser:
      loadBalancer:
        servers:
        - url: http://192.168.1.188:2583

Authnetik does catch it before hand and have me login but it just sends me to Filebrowser's login page.

Either I need to figure out how to properly change Filebrowser's authentication (which seems extremely obtuse and not well documented) or figure out some way for Authentik to send JWT Tokens.

**Reply by kekec777** [reply]
Hi, I think I got it working but cannot send my settings right now, because I'm out.

I will paste my settings either later today tomorrow :)  gmt +1

**Comment by kekec777** [comment]
So here's my setup for filebrowser :

`  
routers:
 
   filebrowser:
      entryPoints:
        - "websecure"
      rule: "Host(`filebrowser.MYDOMAIN.XXX`)"
      middlewares:
        - auth
      tls:
        certResolver: http
      priority: 10
      service: filebrowser

    filebrowserAUTH:
      entryPoints:
        - "websecure"
      rule: "Host(`filebrowser.MYDOMAIN.XXX`) && PathPrefix(`/outpost.goauthentik.io/`)"
      priority: 15
      tls:
        certResolver: http
      service: authentikservice

#######

  services:
    filebrowser:
      loadBalancer:
        servers:
          - url: "http://192.168.1.100:3670"

    authentikservice:
      loadBalancer:
        servers:
          - url: http://authentikserver:9000/outpost.goauthentik.io/

  middlewares:
    auth:
      forwardauth:
        address: http://192.168.1.198:9000/outpost.goauthentik.io/auth/traefik
        #http://authentikserver:9000/outpost.goauthentik.io/auth/traefik
        trustForwardHeader: true
        authResponseHeaders:
          - X-authentik-username
          - X-authentik-groups
          - X-authentik-email
          - X-authentik-name
          - X-authentik-uid
          - X-authentik-jwt
          - X-authentik-meta-jwks
          - X-authentik-meta-outpost
          - X-authentik-meta-provider
          - X-authentik-meta-app
          - X-authentik-meta-version
          - X-Transmission-Session-Id
           #        - authorization # for http user and password  authentik  (Send http-basic authentintic)

`

I hope this help you ... 👍
